pages/events/[slug].tsx

```diff
@@ -55,7 +55,25 @@ const Events = (props: EventType) => {
   )
 }
 
-export async function getServerSideProps(context: { params: { slug: any } }) {
+export async function getStaticPaths() {
+  let datas: EventType[]
+  try {
+    datas = await allEvents()
+  } catch (err) {
+    throw err
+  }
+  const paths = datas.map((data) => {
+    return {
+      params: { slug: data.slug },
+    }
+  })
```

Comment:
yes, this works perfectly fine. But is naming is also important, using `events` or something similar makes more sense!
```suggestion
  let events: EventType[]
  try {
    events = await allEvents()
  } catch (err) {
    throw err
  }
  const paths = events.map(({ slug }) => {
    return {
      params: { slug },
    }
  })
```